### ISSUE TITLE
"deno info" should ignore lock file

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#17169]
Currently if there's a "deno.json(c)" and "deno.lock" then "deno info" will print out all npm packages that are listed in the lockfile, even if the entry point is not an npm package (nor does it have npm imports). This is unexpected behavior and led to some confusion in https://github.com/denoland/deno/issues/17167#issuecomment-1363397034.

Since `deno info` doesn't have `--lock` flag available in this case the lockfile should be ignored.

**Comment by dsherret** [comment]
I don't think it should ignore the lockfile because the lockfile may contain information such as what version a package requirement should resolve to. It seems like there's two things we should fix here:

1. `deno info` should have the lock flags.
2. Only show npm packages relevant to the files that were analyzed.

**Comment by bartlomieju** [comment]
I don't think this is relevant anymore.
